microindex/trailer.go

```diff
@@ -87,7 +87,7 @@ func readTrailer(r io.ReadSeeker, n int64) (*Trailer, int, error) {
 		return nil, 0, err
 	}
 	buf := make([]byte, n)
-	cc, err := r.Read(buf)
+	cc, err := io.ReadFull(r, buf)
```

Comment:
Nit: When io.ReadFull returns a nil error, cc is guaranteed to be len(buf), so you don't need to check cc below.